### PR TITLE
Update the transcript parser to not bomb for "help"/"?"/"help commnad"/"help-topics"

### LINKED ISSUE #1433: `help` bombs in TrascriptParser
At the moment, an attempt to use any of the following commands is not supported by the transcript parser, due to them not being handled by the input handler, but during the main loop:
* `help`
* `?`
* `help (existing command)`
* `help-topics`

### PR DESCRIPTION
**Pull request by pete-ts**
## Overview

A `ucm` block that contains a `help`,`help x`, `?` and `help-topics` directive causes the TranscriptParser to exit with a "die" after printing the failing stanza.

This PR changes this behaviour and allows `help` and friends to have their output printed.

The first two commits (if cloned and run) showcase the breakage. 

Fix #1433

## Implementation notes

When anything help-related is entered by the user, it gets picked by main, parsed and comes back as a `Left` during the `ucm` input reading loop. The same process is performed by the transcript parser in the section of code I am updating.

## Test coverage

Transcript provided that tests all 4 covered scenarios.

## Loose ends

N/A

### CODE REVIEW

File: unison-src/transcripts/help.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+```ucm
+.> help push
+.> help
+.> ?
+.> help-topics
```
Comment (pete-ts) [inline unison-src/transcripts/help.md:5]: all valid scenarios covered (that I am aware of at least; please comment if I am missing something.)

File: parser-typechecker/src/Unison/Codebase/TranscriptParser.hs [hunk 2]
```diff
@@ -175,11 +177,16 @@ run dir configFile stanzas codebase = do
...
-                    Left msg -> do
-                      output $ P.toPlain 65 (P.indentN 2 msg <> P.newline <> P.newline)
-                      dieWithMsg
-                    Right input -> pure $ Right input
+                  Just pat -> 
+                    case IP.parse pat args of
+                      Left msg -> do
+                        output $ P.toPlain 65 (P.indentN 2 msg <> P.newline <> P.newline)
+                        let commands = IP.patternName <$> IPS.validInputs
+                        if IP.patternName pat `elem` ["help", "?", "help-topics"]
```
Comment (aryairani) [inline parser-typechecker/src/Unison/Codebase/TranscriptParser.hs:185]: Better to reference the fields in `IPS.help` and `IPS.helpTopics` than to hard-code strings here

File: unison-src/transcripts/help.output.md [hunk 3]
```diff
@@ -0,0 +1,726 @@
...
+  `push remote`        publishes the current namespace into the
+                       remote namespace `remote`
+  `push`               publishes the current namespace into the
+                       remote namespace configured in
+                       `.unisonConfig` with the key `GitUrl.ns`
+                       where `ns` is the current namespace
+  where `remote` is a git repository, optionally followed by `:`
+  and an absolute remote path, such as:
+  `https://github.com/org/repo`
+  `https://github.com/org/repo:.some.remote.path`
```
Comment (pete-ts) [inline unison-src/transcripts/help.output.md:18]: Can now TDD the fix for #949

File: parser-typechecker/src/Unison/Codebase/TranscriptParser.hs (hunk 2 above)
Comment (pete-ts) [reply parser-typechecker/src/Unison/Codebase/TranscriptParser.hs:185]: Good shout.

**Comment by pete-ts** [comment]
I can add `HelpI` -> `InputHandler` -> `Output`, I was unclear if there was a limitation around specifically avoiding it. Let me know what your preference is, this is more along the lines of "a quick change to the TranscriptParser", whereas indeed the issue can be resolved by "handling help as input."

**Comment by aryairani** [comment]
Yeah I'm a little on the fence about the design, maybe something like:

In `Input.hs`:
```haskell
data Input =
  -- ...
  | HelpI (Maybe InputPattern)
  | HelpTopicI (Maybe HelpTopic)
```

In `HelpTopics.hs` (new):
```haskell
data HelpTopic = TestCache | FileStatus | NameSpaces | DisallowedAbsolute

toPretty :: HelpTopic -> Pretty ColorText
toPretty = \case
  TestCache -> P.callout "🎈" . P.lines $ [
     -- ...
     -- moved from InputPatterns.hs
```

In `Output.hs`, some pass-through kind of thing, I guess;

In `OutputMessages.hs` print the messages out of `HelpTopics.hs`

WDYT? :)

**Comment by pete-ts** [comment]
I agree and like it better. And ultimately, if fixing via `HelpI` etc, the change in the TranscriptParser is 100% redundant/irrelevant code, so closing this PR, will attempt it via changing it to input handling :+1:

**Pull request closed without merging**